File: bubble_sort/BubbleSort.java
public class BubbleSort {

    private static void sort(int[] list){
        if(list == null || list.length == 0 || list.length == 1)
            return;
        boolean swapped;
        int j = 1;
        do{
            swapped = false;
            for (int i = 0; i < list.length - j; i++) {
                if(list[i] > list[i+1]){
                    int temp = list[i];
                    list[i] = list[i+1];
                    list[i+1] = temp;
                    swapped = true;
                }
            }
            j++;
        }
        while (swapped);
    }

    public static void main(String[] args){
        int[] list = { 8, 4, 6, 1, 10, 7, 2, 5, 9, 3 };
        sort(list);
        for (int a : list) {
            System.out.print(a + " ");
        }
    }
}

File: bubble_sort/bubble_sort.py
def bubble_sort(arr):
    """
    Performs a bubble sort
    :type arr: int
    """
    for index_1 in range(len(arr)):
        for index_2 in range(0,len(arr)-index_1-1):
            if(arr[index_2]>arr[index_2+1]):
                temp=arr[index_2+1]
                arr[index_2+1]=arr[index_2]
                arr[index_2]=temp


def main():
    arr = [2, 3, 0, 4,-4,56,12]
    print("Before Sorting")
    bubble_sort(arr)
    print(arr)

    
if __name__ == '__main__':
    main()
